test/framework-ondata-mocha.js

```diff
@@ -714,5 +714,223 @@ describe("framework-ondata", function() {
             }).should.eventually.deep.equal(payload);
         });
 
+        it("should propagate input metadata downstream", function() {
+            var testId = "test identifier";
+            var testdata = "more data";
+            return test.createNetwork({
+                node1: {getComponent: factory({inPorts: {input: {}}, 
+                                               outPorts: {output: {}}
+                                             },
+                                             {fRunUpdater: function(vni){ 
+                                                  vni.outputState({id: testId, data: "some data", lm: createLm()}); 
```

Comment:
No, that's fine.  I meant: please add a comment to the test saying:
// Set metadata property called 'id'

I am suggesting this only to make it easier to see.  when I first looked, it wasn't clear to me what metadata was being set, because it relies on the user knowing that 'id' is not one of the framework properties.  Actually, now that I think of it, maybe it would a little clearer if it were called 'myId' instead of 'id'.  It would stand out more.